spring-batch-docs/asciidoc/whatsnew.adoc

```diff
@@ -4,69 +4,151 @@
+
+* A connection factory to acquire connections from the messaging middleware (JMS, AMQP, and others)
+* A `MessagingTemplate` to send requests from the master to the workers and back again
+* An input channel and an output channel for Spring Integration to get messages from the messaging middleware
+* A special item writer (`ChunkMessageChannelItemWriter`) on the master side that knows how to send chunks of data to workers for processing and writing
+* A message listener (`ChunkProcessorChunkHandler`) on the worker side to receive data from the master
+
+This can be a bit daunting at first glance. This release introduces a new annotation
+named `@EnableBatchIntegration` as well as new APIs (`RemoteChunkingMasterStepBuilder`
+and `RemoteChunkingWorkerBuilder`) to simplify the configuration. The following
+example shows how to use the new annotation and APIs:
+
+[source, java]
+----
+@Configuration
+@EnableBatchProcessing
+@EnableBatchIntegration
+public class RemoteChunkingAppConfig {
+
+   @Autowired
+   private RemoteChunkingMasterStepBuilderFactory masterStepBuilderFactory;
+
+   @Autowired
+   private RemoteChunkingWorkerBuilder workerBuilder;
+
+   @Bean
+   public TaskletStep masterStep() {
+         return this.masterStepBuilderFactory
+         	        .get("masterStep")
+         	        .chunk(100)
+         	        .reader(itemReader())
+         	        .outputChannel(outgoingRequestsToWorkers())
+         	        .inputChannel(incomingRepliesFromWorkers())
+         	        .build();
+   }
+
+   @Bean
+   public IntegrationFlow worker() {
+         return this.workerBuilder
+         	        .itemProcessor(itemProcessor())
+         	        .itemWriter(itemWriter())
+         	        .inputChannel(incomingRequestsFromMaster())
+         	        .outputChannel(outgoingRepliesToMaster())
+         	        .build();
+   }
+}
+----
+
+This new annotation and builders take care of the heavy lifting of configuring
+infrastructure beans. You can now easily configure a master step as well as
+a Spring Integration flow on the worker side. You can find a remote chunking sample
+that uses these new APIs in the
+link:$$https://github.com/spring-projects/spring-batch/tree/master/spring-batch-samples#remote-chunking-sample$$[samples module]
+as well as more details in the <<spring-batch-integration.adoc#remote-chunking,Spring Batch Integration>> chapter.
+
+[[whatsNewJson]]
+=== JSON support
+
+Spring Batch 4.1 adds support for the JSON format. This release introduces a new
+item reader that can read a JSON resource in the following format:
+
+[source, json]
+----
+[
+  {
+    "isin": "123",
+    "quantity": 1,
+    "price": 1.2,
+    "customer": "foo"
+  },
+  {
+    "isin": "456",
+    "quantity": 2,
+    "price": 1.4,
+    "customer": "bar"
+  }
+]
+----
+
+Similar to the `StaxEventItemReader` for XML, the new `JsonItemReader` uses streaming
+APIs to read JSON objects in chunks. Two JSON libraries are supported:
```

Comment:
" are supported out of the box:"

Reply:
I'd ditch "out of the box" there. (I've been removing it in general, because it rarely adds any meaning.)

Reply:
I want to communicate that _we_ provide support for those libraries but if you want to add others, you can implement your own...

Reply:
To do that, I'd write:

"Spring Batch supports two libraries:"

Then add the links on the next lines.

Then add, on the line after the links, "To add other libraries, you can implement your own...."

Reply:
ok, I updated the sentence accordingly.

Reply:
Thank you. :)